source/s3_client.c

```diff
@@ -979,12 +999,11 @@ static struct aws_s3_meta_request *s_s3_client_meta_request_factory_default(
                 return NULL;
             }
 
-            struct aws_input_stream *input_stream = aws_http_message_get_body_stream(options->message);
-
-            if ((input_stream == NULL) && (options->send_filepath.len == 0)) {
+            if (body_source_count == 0) {
```

Comment:
do we need similar check for default case?

Reply:
Good question. Let's chat tomorrow